espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/SourceDocument.java

```diff
@@ -0,0 +1,138 @@
+package io.appium.espressoserver.lib.model;
+
+import android.view.View;
+import android.view.ViewGroup;
+
+import org.w3c.dom.*;
+import org.w3c.dom.Element;
+
+import java.io.StringWriter;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import io.appium.espressoserver.lib.handlers.exceptions.XPathLookupException;
+import io.appium.espressoserver.lib.viewaction.RootViewFinder;
+
+public class SourceDocument {
+
+    private Document doc;
+    private Map<Element, View> viewMap = new HashMap<>();
+    private boolean cacheElementReferences;
+    private final DocumentBuilderFactory docFactory = DocumentBuilderFactory.newInstance();
+    private final DocumentBuilder docBuilder = docFactory.newDocumentBuilder();
+    private TransformerFactory transformerFactory = TransformerFactory.newInstance();
+    private Transformer transformer;
+
+    public SourceDocument() throws ParserConfigurationException, TransformerException {
+        init();
+        buildXML(false);
+    }
+
+    private SourceDocument(boolean cacheElementReferences) throws ParserConfigurationException, TransformerException {
+        init();
+        buildXML(cacheElementReferences);
+    }
+
+    private void init() throws TransformerException {
+        transformer = transformerFactory.newTransformer();
+        transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+        transformer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "2");
+    }
+    /**
+     * Recursively visit all of the views and map them to XML elements
+     * @param cacheElementReferences Keep references of element's mapped to views
+     * @throws ParserConfigurationException
+     */
+    private void buildXML(boolean cacheElementReferences) throws ParserConfigurationException {
+        // Create an empty document
+        this.cacheElementReferences = cacheElementReferences;
```

Comment:
Why cannot we set this variable directly in the constructor? Then there will be no need to pass it method argument.